### PR TITLE
Enforce levelization in libxrpl with CMake

### PR DESCRIPTION
**Pull request by thejohnfreeman**
Adds two CMake functions:

- `add_module(library subdirectory)`: Declares an `OBJECT` "library" (a CMake abstraction for a collection of object files) with sources from the given subdirectory of the given library, representing a **module.** Isolates the module's headers by creating a subdirectory in the build directory, e.g. `.build/tmp123`, that contains just a symlink, e.g. `.build/tmp123/basics`, to the module's header directory, `e.g. include/xrpl/basics`, in the source directory, and putting `.build/tmp123` (but not `include/xrpl`) on the include path of the module sources. This prevents the module sources from including headers not explicitly linked to the module in CMake with `target_link_libraries`.
- `target_link_modules(library scope modules...)`: Links the library target to each of the module targets, and removes their sources from its source list (so they are not compiled and linked twice).

Uses these functions to separate and explicitly link modules in libxrpl:

- Level 01: `beast`
- Level 02: `basics`
- Level 03: `json`, `crypto`
- Level 04: `protocol`
- Level 05: `resource`, `server`

### CODE REVIEW

**Comment by ximinez** [comment]
Not a review yet, but this is such a cool idea.

File: cmake/isolate_headers.cmake
Comment (Bronek) [inline cmake/isolate_headers.cmake:40]: I think `/include/` is misleading here. It's source file isolation feature, not includes.

Comment (Bronek) [reply cmake/isolate_headers.cmake:40]: I suggest
```
-  set(X "${CMAKE_CURRENT_BINARY_DIR}/include/${target}")
+  set(X "${CMAKE_CURRENT_BINARY_DIR}/modules/${target}")
```

File: cmake/add_module.cmake
Comment (Bronek) [inline cmake/add_module.cmake:36]: I think this is not needed ? Or my misunderstanding of how CMake works just showed :-D

File: cmake/RippledCore.cmake
Comment (Bronek) [inline cmake/RippledCore.cmake:126]: I think this is probably not needed ?

Comment (Bronek) [reply cmake/RippledCore.cmake:126]: actually, upon some testing, this whole `target_include_directories` seem to be not needed.

**Comment by Bronek** [comment]
I like how you moved `extract` to appropriate locations, but it seems that `extract` is function-wise the same as `hash`.

Given we already specialize both `boost::hash` (e.g. in `IPEndpoint.h`) and `std::hash` (e.g. in `Book.h`) I think that cleaning this up belongs in a different PR (also converting `extract` to `hash`), so maybe just take a note for future refactoring.

File: cmake/add_module.cmake
Comment (thejohnfreeman) [reply cmake/add_module.cmake:36]: It's in case a module has private headers, not installed, but included by other translation units. I don't know if any of our modules have these (yet), but I'm copying this implementation from my more generalized project.

File: cmake/RippledCore.cmake
Comment (thejohnfreeman) [reply cmake/RippledCore.cmake:126]: It's not needed now that all of the subdirectories in libxrpl are modules. It was necessary as I migrated them one-by-one, and it could be necessary in the future if someone adds a non-module subdirectory. Could be nice if it "just works" without tampering with the CMake. What do you think?

Comment (Bronek) [reply cmake/RippledCore.cmake:126]: Perhaps comment out and put a comment above, to explain why this is not necessary and in what condition it might become necessary again ?

**Comment by Bronek** [comment]
I do wonder whether `Builds/levelization` is becoming superfluous with this change. Why should we keep it, if the ordering of modules (and header isolation) enforce it for us ?

File: cmake/add_module.cmake
Comment (Bronek) [reply cmake/add_module.cmake:36]: Good point. I can see how it might become useful as we move more modules from `xrpld` to `libxrpl` .

File: include/xrpl/protocol/XRPAmount.h
Comment (Bronek) [inline include/xrpl/protocol/XRPAmount.h:1]: Nice work moving the file to where it belongs and reversing the `Number` dependency

File: include/xrpl/protocol/XRPAmount.h
Comment (Bronek) [inline include/xrpl/protocol/XRPAmount.h:161]: Would `explicit` be useful here ?

File: src/libxrpl/basics/partitioned_unordered_map.cpp
Comment (Bronek) [inline src/libxrpl/basics/partitioned_unordered_map.cpp:1]: :1st_place_medal:

**Review by Bronek — COMMENTED** [review]
Overall very nice change. Some comments in the middle, unsure if they require changes in this PR or not.

File: include/xrpl/protocol/XRPAmount.h
Comment (thejohnfreeman) [reply include/xrpl/protocol/XRPAmount.h:161]: I will try that.

**Comment by thejohnfreeman** [comment]
`Builds/levelization` is still used to diagnose cycles in xrpld. I want to remove it after we've eliminated all of the cycles.

**Comment by Bronek** [comment]
> `Builds/levelization` is still used to diagnose cycles in xrpld. I want to remove it after we've eliminated all of the cycles.

Makes sense. All the more reason to make an effort moving parts of `xrpld` into `libxrpl`, but that's separate discussion and more than one PR.

File: include/xrpl/protocol/XRPAmount.h
Comment (thejohnfreeman) [reply include/xrpl/protocol/XRPAmount.h:161]: There are too many call sites depending on the implicit conversion.

**Comment by ximinez** [comment]
Bad news right out the gate on Windows builds:
```
$ cmake build/cmake/msvc19.ON
-- Using Conan toolchain: C:/Dev/rippled/review1/build/conan.msvc19/build/generators/conan_toolchain.cmake
-- Conan toolchain: C++ Standard 20 with extensions OFF
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045.
[...]
-- Conan: Component target declared 'xxHash::xxhash'
-- Conan: Including build module from 'C:/.conan/4b4fe1f3/1/lib/cmake/protobuf/protobuf-generate.cmake'
-- Conan: Including build module from 'C:/.conan/4b4fe1f3/1/lib/cmake/protobuf/protobuf-module.cmake'
-- Conan: Including build module from 'C:/.conan/4b4fe1f3/1/lib/cmake/protobuf/protobuf-options.cmake'
CMake Error at cmake/isolate_headers.cmake:44 (file):
  file failed to create symbolic link
  'C:/Dev/rippled/review1/build/cmake/msvc19.ON/modules/xrpl.libxrpl.beast/xrpl/beast':
  A required privilege is not held by the client.

Call Stack (most recent call first):
  cmake/add_module.cmake:25 (isolate_headers)
  cmake/RippledCore.cmake:69 (add_module)
  CMakeLists.txt:123 (include)


CMake Error at cmake/isolate_headers.cmake:44 (file):
  file failed to create symbolic link
  'C:/Dev/rippled/review1/build/cmake/msvc19.ON/modules/xrpl.libxrpl.basics/xrpl/basics':
  A required privilege is not held by the client.

Call Stack (most recent call first):
  cmake/add_module.cmake:25 (isolate_headers)
  cmake/RippledCore.cmake:76 (add_module)
  CMakeLists.txt:123 (include)


CMake Error at cmake/isolate_headers.cmake:44 (file):
  file failed to create symbolic link
  'C:/Dev/rippled/review1/build/cmake/msvc19.ON/modules/xrpl.libxrpl.json/xrpl/json':
  A required privilege is not held by the client.

Call Stack (most recent call first):
  cmake/add_module.cmake:25 (isolate_headers)
  cmake/RippledCore.cmake:80 (add_module)
  CMakeLists.txt:123 (include)


CMake Error at cmake/isolate_headers.cmake:44 (file):
  file failed to create symbolic link
  'C:/Dev/rippled/review1/build/cmake/msvc19.ON/modules/xrpl.libxrpl.crypto/xrpl/crypto':
  A required privilege is not held by the client.

Call Stack (most recent call first):
  cmake/add_module.cmake:25 (isolate_headers)
  cmake/RippledCore.cmake:83 (add_module)
  CMakeLists.txt:123 (include)


CMake Error at cmake/isolate_headers.cmake:44 (file):
  file failed to create symbolic link
  'C:/Dev/rippled/review1/build/cmake/msvc19.ON/modules/xrpl.libxrpl.protocol/xrpl/protocol':
  A required privilege is not held by the client.

Call Stack (most recent call first):
  cmake/add_module.cmake:25 (isolate_headers)
  cmake/RippledCore.cmake:87 (add_module)
  CMakeLists.txt:123 (include)


CMake Error at cmake/isolate_headers.cmake:44 (file):
  file failed to create symbolic link
  'C:/Dev/rippled/review1/build/cmake/msvc19.ON/modules/xrpl.libxrpl.resource/xrpl/resource':
  A required privilege is not held by the client.

Call Stack (most recent call first):
  cmake/add_module.cmake:25 (isolate_headers)
  cmake/RippledCore.cmake:94 (add_module)
  CMakeLists.txt:123 (include)


CMake Error at cmake/isolate_headers.cmake:44 (file):
  file failed to create symbolic link
  'C:/Dev/rippled/review1/build/cmake/msvc19.ON/modules/xrpl.libxrpl.server/xrpl/server':
  A required privilege is not held by the client.

Call Stack (most recent call first):
  cmake/add_module.cmake:25 (isolate_headers)
  cmake/RippledCore.cmake:97 (add_module)
  CMakeLists.txt:123 (include)


-- xrpl.libxrpl.basics with 14 sources took 14 sources from xrpl.libxrpl
-- xrpl.libxrpl.beast with 15 sources took 15 sources from xrpl.libxrpl
-- xrpl.libxrpl.crypto with 3 sources took 3 sources from xrpl.libxrpl
-- xrpl.libxrpl.json with 9 sources took 9 sources from xrpl.libxrpl
-- xrpl.libxrpl.protocol with 51 sources took 51 sources from xrpl.libxrpl
-- xrpl.libxrpl.resource with 4 sources took 4 sources from xrpl.libxrpl
-- xrpl.libxrpl.server with 2 sources took 2 sources from xrpl.libxrpl
-- Configuring incomplete, errors occurred!
```

**Comment by thejohnfreeman** [comment]
@ximinez ok, try again.

**Comment by ximinez** [comment]
It's building successfully now. Neat workaround. Super annoying that this is an issue, though.

I remember there were some `file(CREATE_LINK` calls elsewhere that were simply disabled under Windows. Would it be worth coming back to those to use this same technique?

Otherwise, I'll come back to the rest of this review as soon as I get some other priorities knocked out.

File: cmake/add_module.cmake
Comment (godexsoft) [inline cmake/add_module.cmake:15]: Wouldn't it be even nicer if you could optionally pass the libraries to link the target with right here too? Is it possible to add the `PUBLIC lib1 lib2` through args somehow?

**Review by godexsoft — APPROVED** [review]
This looks good 👍 Just one question - not a blocker by any means.

**Comment by ximinez** [comment]
@thejohnfreeman I see this has two approvals. Since I've verified that it builds under Windows, it's up to you if you want to wait for mine, too, or go ahead and add the "Passed" label and provide a commit message. I'll preemptively mark it as "Blocked" now to be sure we wait until after the 2.3.0 release before merging.

File: cmake/add_module.cmake
Comment (thejohnfreeman) [reply cmake/add_module.cmake:15]: I find the current call pattern nice enough. It reads like conventional CMake to me. If someone wants to add the function you're talking about, I think that's fine, but I'd rather keep this function just doing the one thing.

**Comment by thejohnfreeman** [comment]
@ximinez I would still appreciate your review.

**Review by ximinez — APPROVED** [review]
Looks good, and it all works!
```
$ find install -type f
install/bin/rippled.exe
install/etc/rippled.cfg
install/etc/validators.txt
install/include/ed25519.h
install/include/secp256k1.h
install/include/secp256k1_ecdh.h
install/include/secp256k1_extrakeys.h
install/include/secp256k1_preallocated.h
install/include/secp256k1_schnorrsig.h
install/include/xrpl/basics/algorithm.h
install/include/xrpl/basics/Archive.h
install/include/xrpl/basics/base64.h
install/include/xrpl/basics/base_uint.h
install/include/xrpl/basics/BasicConfig.h
install/include/xrpl/basics/Blob.h
install/include/xrpl/basics/Buffer.h
install/include/xrpl/basics/ByteUtilities.h
install/include/xrpl/basics/chrono.h
install/include/xrpl/basics/comparators.h
install/include/xrpl/basics/CompressionAlgorithms.h
install/include/xrpl/basics/contract.h
install/include/xrpl/basics/CountedObject.h
install/include/xrpl/basics/DecayingSample.h
install/include/xrpl/basics/Expected.h
install/include/xrpl/basics/FileUtilities.h
install/include/xrpl/basics/hardened_hash.h
install/include/xrpl/basics/join.h
install/include/xrpl/basics/KeyCache.h
install/include/xrpl/basics/LocalValue.h
install/include/xrpl/basics/Log.h
install/include/xrpl/basics/make_SSLContext.h
install/include/xrpl/basics/MathUtilities.h
install/include/xrpl/basics/mulDiv.h
install/include/xrpl/basics/Number.h
install/include/xrpl/basics/partitioned_unordered_map.h
install/include/xrpl/basics/random.h
install/include/xrpl/basics/RangeSet.h
install/include/xrpl/basics/README.md
install/include/xrpl/basics/Resolver.h
install/include/xrpl/basics/ResolverAsio.h
install/include/xrpl/basics/safe_cast.h
install/include/xrpl/basics/scope.h
install/include/xrpl/basics/SHAMapHash.h
install/include/xrpl/basics/SlabAllocator.h
install/include/xrpl/basics/Slice.h
install/include/xrpl/basics/spinlock.h
install/include/xrpl/basics/strHex.h
install/include/xrpl/basics/StringUtilities.h
install/include/xrpl/basics/TaggedCache.h
install/include/xrpl/basics/tagged_integer.h
install/include/xrpl/basics/ToString.h
install/include/xrpl/basics/UnorderedContainers.h
install/include/xrpl/basics/UptimeClock.h
install/include/xrpl/beast/asio/io_latency_probe.h
install/include/xrpl/beast/clock/abstract_clock.h
install/include/xrpl/beast/clock/basic_seconds_clock.h
install/include/xrpl/beast/clock/manual_clock.h
install/include/xrpl/beast/container/aged_container.h
install/include/xrpl/beast/container/aged_container_utility.h
install/include/xrpl/beast/container/aged_map.h
install/include/xrpl/beast/container/aged_multimap.h
install/include/xrpl/beast/container/aged_multiset.h
install/include/xrpl/beast/container/aged_set.h
install/include/xrpl/beast/container/aged_unordered_map.h
install/include/xrpl/beast/container/aged_unordered_multimap.h
install/include/xrpl/beast/container/aged_unordered_multiset.h
install/include/xrpl/beast/container/aged_unordered_set.h
install/include/xrpl/beast/container/detail/aged_associative_container.h
install/include/xrpl/beast/container/detail/aged_container_iterator.h
install/include/xrpl/beast/container/detail/aged_ordered_container.h
install/include/xrpl/beast/container/detail/aged_unordered_container.h
install/include/xrpl/beast/container/detail/empty_base_optimization.h
install/include/xrpl/beast/core/CurrentThreadName.h
install/include/xrpl/beast/core/LexicalCast.h
install/include/xrpl/beast/core/List.h
install/include/xrpl/beast/core/LockFreeStack.h
install/include/xrpl/beast/core/SemanticVersion.h
install/include/xrpl/beast/hash/hash_append.h
install/include/xrpl/beast/hash/uhash.h
install/include/xrpl/beast/hash/xxhasher.h
install/include/xrpl/beast/insight/Collector.h
install/include/xrpl/beast/insight/Counter.h
install/include/xrpl/beast/insight/CounterImpl.h
install/include/xrpl/beast/insight/Event.h
install/include/xrpl/beast/insight/EventImpl.h
install/include/xrpl/beast/insight/Gauge.h
install/include/xrpl/beast/insight/GaugeImpl.h
install/include/xrpl/beast/insight/Group.h
install/include/xrpl/beast/insight/Groups.h
install/include/xrpl/beast/insight/Hook.h
install/include/xrpl/beast/insight/HookImpl.h
install/include/xrpl/beast/insight/Insight.h
install/include/xrpl/beast/insight/Meter.h
install/include/xrpl/beast/insight/MeterImpl.h
install/include/xrpl/beast/insight/NullCollector.h
install/include/xrpl/beast/insight/StatsDCollector.h
install/include/xrpl/beast/net/IPAddress.h
install/include/xrpl/beast/net/IPAddressConversion.h
install/include/xrpl/beast/net/IPAddressV4.h
install/include/xrpl/beast/net/IPAddressV6.h
install/include/xrpl/beast/net/IPEndpoint.h
install/include/xrpl/beast/rfc2616.h
install/include/xrpl/beast/test/yield_to.h
install/include/xrpl/beast/type_name.h
install/include/xrpl/beast/unit_test/amount.h
install/include/xrpl/beast/unit_test/detail/const_container.h
install/include/xrpl/beast/unit_test/global_suites.h
install/include/xrpl/beast/unit_test/match.h
install/include/xrpl/beast/unit_test/recorder.h
install/include/xrpl/beast/unit_test/reporter.h
install/include/xrpl/beast/unit_test/results.h
install/include/xrpl/beast/unit_test/runner.h
install/include/xrpl/beast/unit_test/suite.h
install/include/xrpl/beast/unit_test/suite_info.h
install/include/xrpl/beast/unit_test/suite_list.h
install/include/xrpl/beast/unit_test/thread.h
install/include/xrpl/beast/unit_test.h
install/include/xrpl/beast/utility/Journal.h
install/include/xrpl/beast/utility/maybe_const.h
install/include/xrpl/beast/utility/PropertyStream.h
install/include/xrpl/beast/utility/rngfill.h
install/include/xrpl/beast/utility/temp_dir.h
install/include/xrpl/beast/utility/WrappedSink.h
install/include/xrpl/beast/utility/Zero.h
install/include/xrpl/beast/xor_shift_engine.h
install/include/xrpl/crypto/csprng.h
install/include/xrpl/crypto/README.md
install/include/xrpl/crypto/RFC1751.h
install/include/xrpl/crypto/secure_erase.h
install/include/xrpl/json/detail/json_assert.h
install/include/xrpl/json/JsonPropertyStream.h
install/include/xrpl/json/json_errors.h
install/include/xrpl/json/json_forwards.h
install/include/xrpl/json/json_reader.h
install/include/xrpl/json/json_value.h
install/include/xrpl/json/json_writer.h
install/include/xrpl/json/Object.h
install/include/xrpl/json/Output.h
install/include/xrpl/json/README.md
install/include/xrpl/json/to_string.h
install/include/xrpl/json/Writer.h
install/include/xrpl/proto/org/xrpl/rpc/v1/get_ledger.grpc.pb.h
install/include/xrpl/proto/org/xrpl/rpc/v1/get_ledger.pb.h
install/include/xrpl/proto/org/xrpl/rpc/v1/get_ledger.proto
install/include/xrpl/proto/org/xrpl/rpc/v1/get_ledger_data.grpc.pb.h
install/include/xrpl/proto/org/xrpl/rpc/v1/get_ledger_data.pb.h
install/include/xrpl/proto/org/xrpl/rpc/v1/get_ledger_data.proto
install/include/xrpl/proto/org/xrpl/rpc/v1/get_ledger_diff.grpc.pb.h
install/include/xrpl/proto/org/xrpl/rpc/v1/get_ledger_diff.pb.h
install/include/xrpl/proto/org/xrpl/rpc/v1/get_ledger_diff.proto
install/include/xrpl/proto/org/xrpl/rpc/v1/get_ledger_entry.grpc.pb.h
install/include/xrpl/proto/org/xrpl/rpc/v1/get_ledger_entry.pb.h
install/include/xrpl/proto/org/xrpl/rpc/v1/get_ledger_entry.proto
install/include/xrpl/proto/org/xrpl/rpc/v1/ledger.grpc.pb.h
install/include/xrpl/proto/org/xrpl/rpc/v1/ledger.pb.h
install/include/xrpl/proto/org/xrpl/rpc/v1/ledger.proto
install/include/xrpl/proto/org/xrpl/rpc/v1/README.md
install/include/xrpl/proto/org/xrpl/rpc/v1/xrp_ledger.grpc.pb.h
install/include/xrpl/proto/org/xrpl/rpc/v1/xrp_ledger.pb.h
install/include/xrpl/proto/org/xrpl/rpc/v1/xrp_ledger.proto
install/include/xrpl/proto/README.md
install/include/xrpl/proto/ripple.pb.h
install/include/xrpl/proto/ripple.proto
install/include/xrpl/protocol/AccountID.h
install/include/xrpl/protocol/AMMCore.h
install/include/xrpl/protocol/AmountConversions.h
install/include/xrpl/protocol/ApiVersion.h
install/include/xrpl/protocol/Asset.h
install/include/xrpl/protocol/Book.h
install/include/xrpl/protocol/BuildInfo.h
install/include/xrpl/protocol/detail/b58_utils.h
install/include/xrpl/protocol/detail/features.macro
install/include/xrpl/protocol/detail/ledger_entries.macro
install/include/xrpl/protocol/detail/secp256k1.h
install/include/xrpl/protocol/detail/sfields.macro
install/include/xrpl/protocol/detail/STVar.h
install/include/xrpl/protocol/detail/token_errors.h
install/include/xrpl/protocol/detail/transactions.macro
install/include/xrpl/protocol/digest.h
install/include/xrpl/protocol/ErrorCodes.h
install/include/xrpl/protocol/Feature.h
install/include/xrpl/protocol/Fees.h
install/include/xrpl/protocol/FeeUnits.h
install/include/xrpl/protocol/HashPrefix.h
install/include/xrpl/protocol/Indexes.h
install/include/xrpl/protocol/InnerObjectFormats.h
install/include/xrpl/protocol/IOUAmount.h
install/include/xrpl/protocol/Issue.h
install/include/xrpl/protocol/json_get_or_throw.h
install/include/xrpl/protocol/jss.h
install/include/xrpl/protocol/Keylet.h
install/include/xrpl/protocol/KeyType.h
install/include/xrpl/protocol/KnownFormats.h
install/include/xrpl/protocol/LedgerFormats.h
install/include/xrpl/protocol/LedgerHeader.h
install/include/xrpl/protocol/messages.h
install/include/xrpl/protocol/MPTAmount.h
install/include/xrpl/protocol/MPTIssue.h
install/include/xrpl/protocol/MultiApiJson.h
install/include/xrpl/protocol/nft.h
install/include/xrpl/protocol/NFTokenID.h
install/include/xrpl/protocol/NFTokenOfferID.h
install/include/xrpl/protocol/nftPageMask.h
install/include/xrpl/protocol/NFTSyntheticSerializer.h
install/include/xrpl/protocol/PayChan.h
install/include/xrpl/protocol/Protocol.h
install/include/xrpl/protocol/PublicKey.h
install/include/xrpl/protocol/Quality.h
install/include/xrpl/protocol/QualityFunction.h
install/include/xrpl/protocol/Rate.h
install/include/xrpl/protocol/README.md
install/include/xrpl/protocol/RippleLedgerHash.h
install/include/xrpl/protocol/RPCErr.h
install/include/xrpl/protocol/Rules.h
install/include/xrpl/protocol/SecretKey.h
install/include/xrpl/protocol/Seed.h
install/include/xrpl/protocol/SeqProxy.h
install/include/xrpl/protocol/serialize.h
install/include/xrpl/protocol/Serializer.h
install/include/xrpl/protocol/SField.h
install/include/xrpl/protocol/Sign.h
install/include/xrpl/protocol/SOTemplate.h
install/include/xrpl/protocol/st.h
install/include/xrpl/protocol/STAccount.h
install/include/xrpl/protocol/STAmount.h
install/include/xrpl/protocol/STArray.h
install/include/xrpl/protocol/STBase.h
install/include/xrpl/protocol/STBitString.h
install/include/xrpl/protocol/STBlob.h
install/include/xrpl/protocol/STCurrency.h
install/include/xrpl/protocol/STExchange.h
install/include/xrpl/protocol/STInteger.h
install/include/xrpl/protocol/STIssue.h
install/include/xrpl/protocol/STLedgerEntry.h
install/include/xrpl/protocol/STNumber.h
install/include/xrpl/protocol/STObject.h
install/include/xrpl/protocol/STParsedJSON.h
install/include/xrpl/protocol/STPathSet.h
install/include/xrpl/protocol/STTx.h
install/include/xrpl/protocol/STValidation.h
install/include/xrpl/protocol/STVector256.h
install/include/xrpl/protocol/STXChainBridge.h
install/include/xrpl/protocol/SystemParameters.h
install/include/xrpl/protocol/TER.h
install/include/xrpl/protocol/tokens.h
install/include/xrpl/protocol/TxFlags.h
install/include/xrpl/protocol/TxFormats.h
install/include/xrpl/protocol/TxMeta.h
install/include/xrpl/protocol/UintTypes.h
install/include/xrpl/protocol/XChainAttestations.h
install/include/xrpl/protocol/XRPAmount.h
install/include/xrpl/resource/Charge.h
install/include/xrpl/resource/Consumer.h
install/include/xrpl/resource/detail/Entry.h
install/include/xrpl/resource/detail/Import.h
install/include/xrpl/resource/detail/Key.h
install/include/xrpl/resource/detail/Kind.h
install/include/xrpl/resource/detail/Logic.h
install/include/xrpl/resource/detail/Tuning.h
install/include/xrpl/resource/Disposition.h
install/include/xrpl/resource/Fees.h
install/include/xrpl/resource/Gossip.h
install/include/xrpl/resource/README.md
install/include/xrpl/resource/ResourceManager.h
install/include/xrpl/resource/Types.h
install/include/xrpl/server/detail/BaseHTTPPeer.h
install/include/xrpl/server/detail/BasePeer.h
install/include/xrpl/server/detail/BaseWSPeer.h
install/include/xrpl/server/detail/Door.h
install/include/xrpl/server/detail/io_list.h
install/include/xrpl/server/detail/JSONRPCUtil.h
install/include/xrpl/server/detail/LowestLayer.h
install/include/xrpl/server/detail/PlainHTTPPeer.h
install/include/xrpl/server/detail/PlainWSPeer.h
install/include/xrpl/server/detail/ServerImpl.h
install/include/xrpl/server/detail/SSLHTTPPeer.h
install/include/xrpl/server/detail/SSLWSPeer.h
install/include/xrpl/server/Handoff.h
install/include/xrpl/server/Port.h
install/include/xrpl/server/Server.h
install/include/xrpl/server/Session.h
install/include/xrpl/server/SimpleWriter.h
install/include/xrpl/server/Writer.h
install/include/xrpl/server/WSSession.h
install/lib/cmake/ed25519/ed25519-targets-debug.cmake
install/lib/cmake/ed25519/ed25519-targets.cmake
install/lib/cmake/libsecp256k1/libsecp256k1-config-version.cmake
install/lib/cmake/libsecp256k1/libsecp256k1-config.cmake
install/lib/cmake/libsecp256k1/libsecp256k1-targets-debug.cmake
install/lib/cmake/libsecp256k1/libsecp256k1-targets.cmake
install/lib/cmake/ripple/RippleConfig.cmake
install/lib/cmake/ripple/RippleConfigVersion.cmake
install/lib/cmake/ripple/RippleTargets-debug.cmake
install/lib/cmake/ripple/RippleTargets.cmake
install/lib/ed25519.lib
install/lib/secp256k1.lib
install/lib/xrpl.lib
install/lib/xrpl.libpb.lib

$ find install -type l
install/bin/xrpld
install/include/ripple
```

**Comment by gregtatcam** [comment]
This commit breaks CLion IDE. Refactoring, break-points in include files, file links to compile errors, intellisense stopped working completely or partially depending on the edited files because the include files copied to `modules` directory and CLion treats those copied include files as the project's includes as opposed to the actual includes. This degrades CLion usability.

**Pull request merged**
